data-prepper-plugins/http-source/src/main/java/org/opensearch/dataprepper/plugins/source/loghttp/LogHTTPService.java

```diff
@@ -91,24 +99,33 @@ private void sendJsonList(List<String> jsonList) throws Exception {
         }
         sb.append("]");
         if (sb.toString().getBytes().length > maxRequestLength) {
+            requestsOverMaximumSizeCounter.increment();
             throw new RuntimeException("Request length "+ sb.toString().getBytes().length + " exceeds maxRequestLength "+ maxRequestLength);
+        } else if (sb.toString().getBytes().length > optimalRequestLength) {
+            requestsOverOptimalSizeCounter.increment();
         }
         buffer.writeBytes(sb.toString().getBytes(), key, bufferWriteTimeoutInMillis);
     }
 
     private HttpResponse processRequest(final AggregatedHttpRequest aggregatedHttpRequest) throws Exception {
         final HttpData content = aggregatedHttpRequest.content();
         List<List<String>> jsonList;
+        boolean jsonListSplitSuccess = false;
```

Comment:
Is "success" the right word here? I had some difficulty understanding this code at first. I think you want to ask is "is the JSON list actually split?".

Perhaps: `isJsonListSplit`?